### ISSUE TITLE
Calculate t0 + 5 and t0 + 10 values to be used in TDM calculation

### ISSUE THREAD

**Issue by jdhoffa** [RMI-PACTA/workflow.data.preparation#31]
The TDM metric was conceived in 2020. At that point, it was convenient to calculate, as t+5 and t+10 had very round numbers (2025 and 2030). 

Now, we would like to continue calculating the TDM metric, however, for 2021Q4 data, this requires calculating scenario targets for t+5 = 2026 and t+10 = 2031. 

This would require making some adjustment to values in the `config.yml` here:
https://github.com/RMI-PACTA/workflow.data.preparation/blob/331f6abf1bbe1ae990960c87a09b8de6a2e63deb/config.yml#L40-L41

The `t+5` case is already handled, however the `t+10` case should be included, perhaps by adding `2031` to the `adittional_year` config. Perhaps it is safer/ more future proof to just add a `tdm_years` configuration parameter, that way we know what we are keeping track of (and can remove it easily if we decide to remove the metric).

We may also consider double-checking WHY we need, for example, the year `2040`.

This is a whole big headache lol.

See further discussion here: https://dev.azure.com/RMI-PACTA/2DegreesInvesting/_workitems/edit/6197

Relates to https://github.com/RMI-PACTA/pacta.portfolio.analysis/pull/221

**Comment by jdhoffa** [comment]
I see a few different options here, and would ask for some input from @cjyetman and/or @AlexAxthelm:

## Option 1:
``` yml
time_horizon: 5
additional_year: [2030, 2031, 2040]
```
This seems the least intrusive, but also likely is introducing tech debt (as I have no idea if we do or do not need 2030 and 2040)

## Option 2:
``` yml
time_horizon: 5
tdm_delta: [5, 10]
additional_year: [2030, 2040]
```
This adds a whole new concept (`tdm_delta`) which might be a bit dangerous but allows us to easily understand in the future WHY we want plus 5 and plus 10 from the start year (which is needed for TDM), and allows us to easily revert back if we decide to remove TDM. 

Would require changes also here:
https://github.com/RMI-PACTA/workflow.data.preparation/blob/331f6abf1bbe1ae990960c87a09b8de6a2e63deb/run_pacta_data_preparation.R#L89-L96

## Option 3:
``` yml
time_horizon: 5
additional_year: [2031, 2040]
```
This option seems unnecessarily dangerous, and also not future-proof... though honestly I have no idea what the values of "additional_year" are even used for. Why do we need "2040"?

**Comment by jdhoffa** [comment]
For the record, my preference here would probably be Option 2.

**Comment by cjyetman** [comment]
definitely prefer option 2, much like we did for `market_share_target_reference_year`

**Comment by cjyetman** [comment]
I believe the `additional_year` was added to support TDM, but I'll have to look that up

**Comment by jdhoffa** [comment]
I don't think so, 2040 was never necessary for that! I believe additional_year is actually a very very legacy concept

**Comment by cjyetman** [comment]
I think you're right... but then TDM piggy-backed on it because it also needed the ten year horizon?

**Comment by jdhoffa** [comment]
Seems like it! Good thing we're fixing that now :-)

**Comment by jdhoffa** [comment]
New issue created to assess what `additional_year` is and if it is or isn't needed.

**Comment by jacobvjk** [comment]
> Maybe 2040 was needed for stress testing @jacobvjk?

see my reply in #33 this parameter was not used for stress testing in the past and I do not know when and why it was introduced

**Comment by AlexAxthelm** [comment]
I agree that calling out the TDM delta years is the right call
